namer/namer.cc

```diff
@@ -38,6 +38,7 @@ enum class DefinitionKind : u1 {
     StaticField = 4,
     TypeMember = 5,
     Symbol = 6,
+    Constant = 7,
```

Comment:
Why are we adding a new DefinitionKind?

Reply:
cc @jvilk if you have a second to look at this and see if this is how you'd implement private constant support or if you'd re-use one of the existing `DefinitionKind`s, that'd be great.

Reply:
@jez @jvilk 
`private_constant` can be called for both a class/module as well as a static field. When handling the `private_constant` call in the `SymbolFinder`, we don't yet know if the argument will refer to a `Class` or a `StaticField`. We don't know that until the `SymbolDefiner` pass. I could maybe repurpose one of the other types in the enum, but I think that would make it harder to reason about.

An alternative could be to stop using `DefinitionKind` in the `Modifier` struct and instead create a new enum for this, it could look something like this:
```
enum class ModifierKind : u1 {
    Class = 0,
    Method = 1,
    ClassOrStaticField = 2,
};
```

Open to suggestions here.

Reply:
@bss after looking at the code a bit, I think your suggestion to introduce a new `ModifierKind` enum is a great idea. That gets at exactly the root of the problem I had with adding a new definition kind—we should never find new "constant" definitions, but it's totally fine for us to modify "any existing definition which is a constant".

If you want to implement that I'd be happy taking this change!

Reply:
fwiw, i would nest that new enum class inside `struct Modifier` so that it's `Modifier::Kind`.